### PR TITLE
Fixes #21684

### PR DESCRIPTION
**Pull request by rominf**

### CODE REVIEW

File: salt/modules/zypper.py [hunk 1]
```diff
@@ -637,8 +636,7 @@ def install(name=None,
     # Split the targets into batches of 500 packages each, so that
     # the maximal length of the command line is not broken
     while targets:
-        cmd = cmd_install
-        cmd.extend(targets[:500])
+        cmd = cmd_install + targets[:500]
```
Comment (isbm) [inline salt/modules/zypper.py]: Here it works, but I would do it explicitly for others, something like:
```python
cmd = cmd_install[:]
cmd.extend(targets[:500])
```

File: salt/modules/zypper.py [hunk 2]
```diff
@@ -628,8 +625,10 @@ def install(name=None,
         log.info('Targeting repo {0!r}'.format(fromrepo))
     else:
         fromrepoopt = ''
-    cmd_install = ['zypper', '--no-refresh', '--non-interactive', 'install',
-                   '--name', '--auto-agree-with-licenses']
+    cmd_install = ['zypper', '--non-interactive']
+    if not refresh:
+        cmd_install.append('--no-refresh')
+    cmd_install += ['install', '--name', '--auto-agree-with-licenses']
```
Comment (isbm) [inline salt/modules/zypper.py]: Looks good now, thanks!

File: salt/modules/zypper.py [hunk 3]
```diff
@@ -655,8 +653,7 @@ def install(name=None,
                 downgrades.append(match.group(1))
 
     while downgrades:
-        cmd = cmd_install + ['--force']
-        cmd.extend(downgrades[:500])
+        cmd = cmd_install + ['--force'] + downgrades[:500]
```
Comment (isbm) [inline salt/modules/zypper.py]: Why you adding `--force` multiple times in a loop?

Comment (isbm) [inline salt/modules/zypper.py]: OK.

File: salt/modules/zypper.py (hunk 1 above)
Comment (rominf) [inline salt/modules/zypper.py]: I know. In fact I don't like calling `zypper` and list manipulations at all (I wrote repository management via python library (https://github.com/saltstack/salt/pull/14158) which you have replaced). So anyway it's a dirty solution IMHO.

**Comment by jfindlay** [comment]
@rominf, @isbm, are you both satisfied with the changes?  Thanks.

**Comment by rominf** [comment]
@jfindlay I'm satisfied.

**Comment by isbm** [comment]
@jfindlay yes, looks good to me.

**Comment by isbm** [comment]
Thanks!

**Pull request merged**